src/Parser/Shaman/Elemental/Modules/Talents/UnlimitedPower.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import Analyzer from 'Parser/Core/Analyzer';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import { formatDuration, formatPercentage } from 'common/format';
+import { STATISTIC_ORDER } from 'Interface/Others/StatisticBox';
+import ExpandableStatisticBox from 'Interface/Others/ExpandableStatisticBox';
+import UnlimitedPowerTimesByStacks from 'Parser/Shaman/Elemental/Modules/Talents/UnlimitedPowerTimesByStacks';
+
+const HASTE_PER_STACK = 0.02;
+
+class UnlimitedPower extends Analyzer {
```

Comment:
Instead of having this split into two classes, you could merge it into one. Example of this implementation:
https://github.com/WoWAnalyzer/WoWAnalyzer/blob/master/src/Parser/Warrior/Protection/Modules/Talents/IntoTheFray.js

PS: Remember to add the haste to haste.js aswell.

Reply:
Will look into that.

Reply:
The BDK one is split into two because it also needs the bone shield stacks for other modules, shouldn't be needed here. Yeah it's fine for now, it's pretty easy to copy the IntoTheFray module though :P